### PR TITLE
Hide `.spacedrive` file on Windows

### LINKED ISSUE #2215: Metadata file is visible on Windows machines
On Windows machines, it has been reported ([example 1](https://discord.com/channels/949090953497567312/949090953950560307/1218060111294238804) | [example 2](https://discord.com/channels/949090953497567312/1183717834749120562)) that the `.spacedrive` metadata file is not hidden by default. This file should not typically be visible except in very advances use cases.

### PR DESCRIPTION
**Pull request by iLynxcat**
This should add the hidden file attribute to `.spacedrive` files on Windows. Need to see if this passes in CI as I don't have a Windows machine to test on.

Closes #2215

### CODE REVIEW

File: core/src/location/metadata.rs [hunk 1]
```diff
@@ -247,13 +250,29 @@ impl SpacedriveLocationMetadataFile {
...
-		.map_err(|e| LocationMetadataError::Write(e, self.path.clone()))
+		let mut file_options = OpenOptions::new();
+
+		// we want to write the file if it exists, otherwise create it
+		file_options.create(true).write(true);
+
+		#[cfg(target_os = "windows")]
+		{
+			use std::os::windows::fs::OpenOptionsExt;
+			file_options.attributes(winapi::FILE_ATTRIBUTES_HIDDEN);
```
Comment (brxken128) [inline core/src/location/metadata.rs:261]: I believe the entire `winapi` crate can be removed (less codegen, faster build times and Windows CI) by replacing `winapi::FILE_ATTRIBUTES_HIDDEN` with `0x00000002` ([as that's what it equals](https://docs.rs/winapi/latest/winapi/um/winnt/constant.FILE_ATTRIBUTE_HIDDEN.html))

Comment (iLynxcat) [reply core/src/location/metadata.rs:261]:
```suggestion
			file_options.attributes(0x00000002); // set file hidden attribute on Windows
```

File: core/src/location/metadata.rs [hunk 2]
```diff
@@ -247,13 +250,30 @@ impl SpacedriveLocationMetadataFile {
...
+
+		let metadata_contents = serde_json::to_vec(&self.metadata)
+			.map_err(|e| LocationMetadataError::Serialize(e, self.path.clone()))?;
+
+		file_options
+			.open(&self.path)
+			.await
+			.map_err(|e| LocationMetadataError::Write(e, self.path.clone()))
+			// TODO: replace unwrap() with proper checks!!!
+			.unwrap()
```
Comment (iLynxcat) [inline core/src/location/metadata.rs:273]: Need somebody better versed in Rust to recommend a better check here

Comment (brxken128) [reply core/src/location/metadata.rs:273]: I can handle this :)

File: core/src/location/metadata.rs [hunk 3]
```diff
@@ -247,13 +250,29 @@ impl SpacedriveLocationMetadataFile {
...
```
Comment (brxken128) [inline core/src/location/metadata.rs:273]:
```suggestion
			.map_err(|e| LocationMetadataError::Write(e, self.path.clone()))?
```

**Comment by brxken128** [comment]
This still could do with being tested on Windows before being merged 🙏🏻

**Comment by brxken128** [comment]
I just pushed the branch into the main repo and started a [release workflow](https://github.com/spacedriveapp/spacedrive/actions/runs/8317945202) in order to make testing this much easier - should be merged very soon if all goes well!

**Review by brxken128 — APPROVED** [review]
Tested on Windows 11 (in a VM) and this works flawlessly! Thank you very much @iLynxcat for the pull request :)

image

**Pull request merged**